macaroonbakery/httpbakery/client.py

```diff
@@ -290,11 +290,20 @@ def hook(response, *args, **kwargs):
     return hook
 
 
-def extract_macaroons(headers):
+def extract_macaroons(headers_or_request):
     ''' Returns an array of any macaroons found in the given slice of cookies.
-    @param headers: dict of headers
+    If headers implements a get_header method, that will be used
+    instead of get to retrieve headers.
```

Comment:
this reads confusingly as far as "instead of get to" this is just retrieving from a dict

Reply:
Rephrased.